src/components/editor/EditorArticleController.jsx

```diff
@@ -92,8 +106,55 @@ export default class EditorArticleController extends FalcorController {
     }, 2000);
   }
 
+  // Handle state changes coming from editAuthorsForm
+
+  handleAddAuthor(slug) {
+    // TODO: Autocomplete which also includes validating that the authors indeed exist
+
+    // Is this author already added?
+    if (!slug || this.state.authorsAdded.hasOwnProperty(slug) || this.state.authorsDeleted.hasOwnProperty(slug)) {
```

Comment:
Why don't you make adding an author inside authorsDeleted just unstrike it and delete the unstrike function.

Then we just have two functions `handleAddAuthor` and `handleDeleteAuthor` and its very simple for a programmer to understand.